GetMessage/userPreferenceChecker/__tests__/userPreferencesCheckerFactory.test.ts

```diff
@@ -0,0 +1,245 @@
+      aServiceId,
+      aFiscalCode
+    )();
+
+    expect(res).toStrictEqual(E.right(O_expectedResult));
+  });
+});
+
+describe("userPreferencesCheckerFactory |> userPreferencesCheckerFactory", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  const f = require("../userPreferencesCheckerFactory");
+
+  // userPreferenceCheckerVersionUNKNOWNToVersionWithReadAuth,
+  //   userPreferenceCheckerVersionWithReadAuth,
+  const spyUnknownImplementation = jest.spyOn(
+    userPreferenceCheckerVersionUNKNOWNToVersionWithReadAuth,
+    "canAccessMessageReadStatus"
+  );
+  const spyVersionGreaterThanImplementation = jest.spyOn(
+    f,
+    "userPreferenceCheckerVersionWithReadAuth"
+  );
+
+  it("should call userPreferenceCheckerVersionWithReadAuth if appVersion is UNKNOWN", async () => {
+    mockServicePreferencesGetter.mockReturnValueOnce(TE.of(O.none));
+
+    const res = await userPreferencesCheckerFactory(
+      {
+        ...aRetrievedProfile,
+        lastAppVersion: "UNKNOWN"
+      },
+      mockServicePreferencesGetter,
+      MIN_READ_STATUS_PREFERENCES_VERSION
+    ).canAccessMessageReadStatus(aServiceId, aFiscalCode)();
+
+    expect(spyUnknownImplementation).toHaveBeenCalled();
+    expect(spyVersionGreaterThanImplementation).not.toHaveBeenCalled();
+
+    expect(res).toStrictEqual(E.right(false));
+  });
+
+  it("should call userPreferenceCheckerVersionWithReadAuth if appVersion is < MIN_READ_STATUS_PREFERENCES_VERSION", async () => {
+    mockServicePreferencesGetter.mockReturnValueOnce(TE.of(O.none));
+
+    const res = await userPreferencesCheckerFactory(
+      {
+        ...aRetrievedProfile,
+        lastAppVersion: PREV_APP_VERIONS
+      },
+      mockServicePreferencesGetter,
+      MIN_READ_STATUS_PREFERENCES_VERSION
+    ).canAccessMessageReadStatus(aServiceId, aFiscalCode)();
+
+    expect(spyUnknownImplementation).toHaveBeenCalled();
+    expect(spyVersionGreaterThanImplementation).not.toHaveBeenCalled();
+
+    expect(res).toStrictEqual(E.right(false));
+  });
+
+  it("should call userPreferenceCheckerVersionUNKNOWNToVersionWithReadAuth if appVersion is = MIN_READ_STATUS_PREFERENCES_VERSION", async () => {
+    mockServicePreferencesGetter.mockReturnValueOnce(TE.of(O.none));
+
+    const res = await userPreferencesCheckerFactory(
+      {
+        ...aRetrievedProfile,
+        lastAppVersion: MIN_READ_STATUS_PREFERENCES_VERSION
+      },
+      mockServicePreferencesGetter,
+      MIN_READ_STATUS_PREFERENCES_VERSION
+    ).canAccessMessageReadStatus(aServiceId, aFiscalCode)();
+
+    expect(spyUnknownImplementation).not.toHaveBeenCalled();
+    expect(spyVersionGreaterThanImplementation).toHaveBeenCalled();
+
+    expect(res).toStrictEqual(E.right(true));
+  });
+
+  it("should call userPreferenceCheckerVersionUNKNOWNToVersionWithReadAuth if appVersion is > MIN_READ_STATUS_PREFERENCES_VERSION", async () => {
```

Comment:
```suggestion
  it("should call userPreferenceCheckerVersionWithReadAuth if appVersion is > MIN_READ_STATUS_PREFERENCES_VERSION", async () => {
```

Reply:
Thanks, done in [74b17c8](https://github.com/pagopa/io-functions-services/pull/211/commits/74b17c8c291dfec61b3dc4255771a58afea77633)